kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.ctl.cmd.log
+
+import scala.collection.JavaConverters._
+
+import org.apache.kyuubi.client.{BatchRestApi, KyuubiRestClient}
+import org.apache.kyuubi.client.api.v1.dto.{Batch, OperationLog}
+import org.apache.kyuubi.ctl.{CliConfig, ClientFactory}
+import org.apache.kyuubi.ctl.cmd.Command
+
+class LogBatchCommand(cliConfig: CliConfig) extends Command(cliConfig) {
+
+  override def validateArguments(): Unit = {
+    if (cliArgs.batchOpts.batchId == null) {
+      fail("Must specify batchId for log batch command.")
+    }
+  }
+
+  override def run(): Unit = {
+    val kyuubiRestClient: KyuubiRestClient = ClientFactory.getKyuubiRestClient(cliArgs, null, conf)
+    val batchRestApi: BatchRestApi = new BatchRestApi(kyuubiRestClient)
+
+    var log: OperationLog = batchRestApi.getBatchLocalLog(
+      cliArgs.batchOpts.batchId,
+      cliArgs.batchOpts.from,
+      cliArgs.batchOpts.size)
+    log.getLogRowSet.asScala.foreach(x => info(x))
+
+    var done = false
+    val batchId = cliArgs.batchOpts.batchId
+    var batch: Batch = null
+    if (cliArgs.logOpts.forward) {
+      while (!done) {
+        log = batchRestApi.getBatchLocalLog(
+          batchId,
+          -1,
```

Comment:
I see, if the from < 0, the server will use the `lastSeekReadPos`.

But to prevent that concurrent conflicts, I think we can calculate the new offset based on last offset + total size.